File: 11-container-with-most-water/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        ans = 0
        while l<=r:
            curr = min(height[l],height[r])*(r-l)
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
            ans = max(ans,curr)
        return ans

File: 11-container-with-most-water/11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int i = 0, j = n-1, ans = min(height[0],height[n-1])*(n-1);
        while(i<j){
            if(height[i]>height[j]){
                ans = max(ans,height[j]*(j-i));
                j--;
            }else{
                ans = max(ans,height[i]*(j-i));
                i++;
            }
        }
        return ans;
    }
};